tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -105,7 +115,119 @@ propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full
+                *    in SqlCommand.cs we have: rpc.rpcName = this.CommandText; // just get the raw command text
+                * Injecting the comment in the CommandText would result in getting an exception that the StoredProcedure name wasn't found (and would look like an empty string).
+                *
+                * We CAN NOT safely do this though if the Command has any parameters that are not Input (e.g. Return, InputOutput, Output).
+                *
+                * What this function does is:
+                *   - Swap to EXEC statement (which is a text command executing a stored procedure).
+                *   - Build a Input parameter list for the EXEC statement, so that we can pass all the parameters to it.
+                * Some helpful links:
+                * reference RPC for stored procedure: https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L5548
+                * reference text SQL (with params): https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L4488
+                * reference text SQL (no params): https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L4437
+                */
+
+                // check to see if we have any Return/InputOutput/Output parameters
+                if (command.Parameters != null)
+                {
+                    foreach (DbParameter? param in command.Parameters)
+                    {
+                        if (param == null)
+                        {
+                            continue;
+                        }
+
+                        if (param.Direction != ParameterDirection.Input)
+                        {
+                            return false;
+                        }
+                    }
+                }
+
+                var procName = command.CommandText ?? string.Empty;
+
+                if (string.IsNullOrEmpty(procName))
+                {
+                    return false;
+                }
+
+                string? quotedName;
+                try
+                {
+                    // dbo.SomeName -> [dbo].[SomeName]
+                    // this uses some underlying SqlClient code that I've copied to parse the identifier
+                    quotedName = VendoredSqlHelpers.ParseAndQuoteIdentifier(procName, isUdtTypeName: false);
+                    if (string.IsNullOrEmpty(quotedName))
+                    {
+                        // if we can't parse the identifier, return false
+                        return false;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex, "Failed to parse/quote stored procedure, no DBM data will be propagated");
```

Comment:
Just flagging that if this errors, it will probably _always_ error, which could lead to a flood of error logs... Maybe we should be safe and rate limit it (e.g. put the `procName` in a `ConcurrentDictionary<>` and see if it's already set? 🤷‍♂️)